pinot-hadoop/src/main/java/com/linkedin/pinot/hadoop/job/mapper/HadoopSegmentCreationMapReduceJob.java

```diff
@@ -205,8 +213,32 @@ protected void map(LongWritable key, Text value, Context context) throws IOExcep
     }
 
     protected void setSegmentNameGenerator(SegmentGeneratorConfig segmentGeneratorConfig, Integer seqId,
-        Path hdfsAvroPath, File dataPath) {
+                                           Path hdfsAvroPath, File dataPath) {
+      Schema schema = segmentGeneratorConfig.getSchema();
+
+      if(schema == null ||
```

Comment:
Can we leave out the incoming granularity spec null check? That is not needed, think that is a malformed schema and should have already failed